model/src/main/scala/uk/gov/ons/addressIndex/model/config/AddressIndexConfig.scala

```diff
@@ -5,7 +5,11 @@ case class ElasticSearchConfig(
   cluster: String,
   uri: String,
   indexes: IndexesConfig,
-  shield: ShieldConfig
+  shield: ShieldConfig,
+  defaultLimit: Int,
+  defaultOffset: Int,
+  maximumLimit: Int,
```

Comment:
Having limits etc. externally configurable proved very useful for WDA (the ONS data API I still provide support for). An example is the maximum number of data cells allowed in the request which could be tuned during performance testing, and also set to different values in different environments. The Elasticsearch documentation talks of performance hits with "deep paging" so this may influence the maximum offset value allowed (although  I do concede that for address matching we're not usually interested in the lower results).